### PR TITLE
Remove deprecated methods and corresponding tests

### PR DESCRIPTION
**Pull request by shinfan**
Update:

Fixed all malformed javadocs including:
- Deprecated method references
- Unqualified class names
- Typos
 

_Note: I intend to merge this PR soon since it may cause lots of conflicts._

Remove deprecated methods and tests in one pass by a self-made script, with a few exceptions:
- There are still plenty of manual work involved to update usages of deprecated methods and fields
- Manual corrections due to misspell of the word "deprecated"
- google-cloud-pubsub: Pubsub has a few deprecated packages and it affects its integrated tests as well. Removing deprecated packages is not in the scope of this PR.
- google-cloud-compute: Compute engine has a few "Deprecated" related service which causes lots of noise of the automated script. 
- google-cloud-datastore
  - In `ValueBuilder` `getExcludeFromIndexes` was marked as deprecated which is inconsistent with other getter/setter methods
  - Preserved the fields `meaning` in `ValueBuilder` since it is still wide used in the code. May require a separate PR to deprecate.

Fixes https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1569

Test: All unit tests passed locally.

### CODE REVIEW

**Comment by shinfan** [comment]
@garrettjonesgoogle one of the Travis test failed (JDK8) but it does not seem to be related to this PR.

**Comment by shinfan** [comment]
Seems like we have several javadocs which refer to the deprecated methods. I will go through the repo and fix them.

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/RawValue.java [hunk 1]
```diff
@@ -84,4 +84,4 @@ static Builder newBuilder(com.google.datastore.v1.Value valuePb) {
     builder.set(valuePb);
     return builder;
   }
-}
```
Comment (garrettjonesgoogle) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/RawValue.java]: undo this change (you are losing the newline at the end of the file)

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/Value.java [hunk 2]
```diff
@@ -250,4 +238,4 @@ public boolean equals(Object obj) {
     return valueType == null ? RawValue.MARSHALLER.fromProto(proto).build()
         : valueType.getMarshaller().fromProto(proto).build();
   }
-}
```
Comment (garrettjonesgoogle) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/Value.java]: add newline back

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/ValueBuilder.java [hunk 3]
```diff
@@ -62,4 +56,4 @@
   B set(V value);
 
   P build();
-}
+}
```
Comment (garrettjonesgoogle) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/ValueBuilder.java]: add newline back

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/MessageReceiver.java [hunk 4]
```diff
@@ -17,6 +17,7 @@
 package com.google.cloud.pubsub.spi.v1;
 
 import com.google.pubsub.v1.PubsubMessage;
+import java.util.concurrent.BlockingQueue;
```
Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/MessageReceiver.java]: Could you remove this import?

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/RawValue.java (hunk 1 above)
Comment (shinfan) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/RawValue.java]: Done

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/Value.java (hunk 2 above)
Comment (shinfan) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/Value.java]: Done

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/ValueBuilder.java (hunk 3 above)
Comment (shinfan) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/ValueBuilder.java]: Done
